src/main/kotlin/org/koitharu/kotatsu/parsers/site/Manhwa18Parser.kt

```
@@ -0,0 +1,180 @@
+		val tagQuery = tags.orEmpty().joinToString(",") { it.key }
+		val url = buildString {
+			append("https://")
+			append(getDomain())
+			append("/manga-list.html?listType=pagination&page=")
+			append(page)
+			append("&artist=&author=&group=&m_status=&name=")
+			append(query.orEmpty())
+			append("&genre=$tagQuery")
+			append("&ungenre=")
+			append("&sort=")
+			append(sortQuery)
+			append("&sort_type=")
+			append(sortType)
+		}
+
+		val docs = context.httpGet(url).parseHtml()
+		val actualPage = docs.selectFirst("ul.pagination  a.active")?.text()?.toIntOrNull()
+		if (actualPage != page) {
+			return emptyList()
+		}
+
+		return docs.select(".card-body .thumb-item-flow")
+			.map {
+				val titleElement = it.selectFirstOrThrow(".thumb_attr.series-title > a")
+				val absUrl = titleElement.attrAsAbsoluteUrl("href")
+				Manga(
+					id = generateUid(absUrl.toRelativeUrl(getDomain())),
+					title = titleElement.text(),
+					altTitle = null,
+					url = absUrl.toRelativeUrl(getDomain()),
+					publicUrl = absUrl,
+					rating = RATING_UNKNOWN,
+					isNsfw = true,
+					coverUrl = it.selectFirst("div.img-in-ratio")?.attrAsAbsoluteUrl("data-bg").orEmpty(),
+					tags = emptySet(),
+					state = null,
+					author = null,
+					largeCoverUrl = null,
+					description = null,
+					source = MangaSource.MANHWA18,
+				)
+			}
+	}
+
+	override suspend fun getPages(chapter: MangaChapter): List<MangaPage> {
+		val chapterUrl = chapter.url.toAbsoluteUrl(getDomain())
+		return context.httpGet(chapterUrl).parseHtml()
+			.select("div.chapter-content > img").map {
+				val url = it.attrAsRelativeUrlOrNull("src").orEmpty()
+				MangaPage(
+					id = generateUid(url),
+					url = url,
+					referer = chapterUrl,
+					preview = null,
+					source = MangaSource.MANHWA18,
+				)
+			}
+	}
+
+	override suspend fun getTags(): Set<MangaTag> {
+		return context.httpGet("https://${getDomain()}/").parseHtml().selectFirst(".genres-menu")
```

Maybe it should be better to throw an exception instead of returning an empty set?
```suggestion
		return context.httpGet("https://${getDomain()}/").parseHtml().selectFirstOrThrow(".genres-menu")
```